index.js

```diff
@@ -58,6 +58,122 @@ class DbFacility extends Base {
       }
     ], cb)
   }
+
+  /**
+   * @see https://www.npmjs.com/package/mysql#pooling-connections
+   * @see https://www.npmjs.com/package/mysql#transactions
+   *
+   * @param {(conn: { cli: mysql.PoolConnection }, txFuncCb: (err: Error|undefined) => void) => void} func
+   * @param {(err: Error|undefined) => void} cb
+   */
+  runTransaction (func, cb) {
+    /** @type {mysql.PoolConnection} */
+    let conn = null
+    let txStarted = false
+    let txCommited = false
+
+    async.series({
+      conn: (next) => {
+        this.cli.getConnection((err, res) => {
+          if (err) return next(err)
+
+          conn = res // intentional
+          return next()
+        })
+      },
+      begin: (next) => {
+        conn.beginTransaction(next)
+      },
+      execute: (next) => {
+        txStarted = true
+        func({ cli: conn }, next)
```

Comment:
We should explicitly document this function to not supported nested transactions otherwise if `func` starts a nested transaction without properly committing it or rolling back, it can mess up the whole flow.